geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/data/DeltaDUnitTest.java

```diff
@@ -0,0 +1,339 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import redis.clients.jedis.Jedis;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.partition.PartitionRegionHelper;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.test.awaitility.GeodeAwaitility;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.dunit.rules.RedisClusterStartupRule;
+
+public class DeltaDUnitTest {
+
+  @ClassRule
+  public static RedisClusterStartupRule clusterStartUp = new RedisClusterStartupRule(4);
+
+  private static final String LOCAL_HOST = "127.0.0.1";
+  private static final int SET_SIZE = 10;
+  private static final int JEDIS_TIMEOUT =
+      Math.toIntExact(GeodeAwaitility.getTimeout().toMillis());
+  private static Jedis jedis1;
+  private static Jedis jedis2;
+
+  private static Properties locatorProperties;
+
+  private static MemberVM locator;
+  private static MemberVM server1;
+  private static MemberVM server2;
+
+  private static int redisServerPort1;
+  private static int redisServerPort2;
+
+  @BeforeClass
+  public static void classSetup() {
+    locatorProperties = new Properties();
+    locatorProperties.setProperty(MAX_WAIT_TIME_RECONNECT, "15000");
+
+    locator = clusterStartUp.startLocatorVM(0, locatorProperties);
+    server1 = clusterStartUp.startRedisVM(1, locator.getPort());
+    server2 = clusterStartUp.startRedisVM(2, locator.getPort());
+
+    redisServerPort1 = clusterStartUp.getRedisPort(1);
+    redisServerPort2 = clusterStartUp.getRedisPort(2);
+
+    jedis1 = new Jedis(LOCAL_HOST, redisServerPort1, JEDIS_TIMEOUT);
+    jedis2 = new Jedis(LOCAL_HOST, redisServerPort2, JEDIS_TIMEOUT);
+  }
+
+  @Before
+  public void testSetup() {
+    jedis1.flushAll();
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    jedis1.disconnect();
+    jedis2.disconnect();
+
+    server1.stop();
+    server2.stop();
+  }
+
+  @Test
+  public void shouldCorrectlyPropagateDeltaToSecondaryServer_whenAppending() {
+    String key = "key";
+    String baseValue = "value-";
+    jedis1.set(key, baseValue);
+    for (int i = 0; i < SET_SIZE; i++) {
+      jedis1.set(key, String.valueOf(i));
+
+      String server1LocalValue = server1.invoke(() -> {
+        InternalCache cache = ClusterStartupRule.getCache();
+        Region<ByteArrayWrapper, RedisData> region = cache.getRegion("__REDIS_DATA");
+        Region<ByteArrayWrapper, RedisData> localRegion =
+            PartitionRegionHelper.getLocalData(region);
+
+        RedisData localValue = localRegion.get(new ByteArrayWrapper(key.getBytes()));
+        return localValue.toString();
```

Comment:
Instead of using the RedisData's toString (which might not show all fields) try converting it to a byte array. You can use BlobHelper.serializeToBlob and then have this return the byte array. Then use the JDK Arrays.equals(byte[], byte[]) to see if they are equal. If they are not you will want to use BlobHelper.deserializeBlob(byte[]) to report the objects that are not equal.

Reply:
good call, updating it!